app/grandchallenge/core/utils/markdown.py

```diff
@@ -9,22 +12,64 @@ def extendMarkdown(self, md):  # noqa: N802
 
 
 class BS4Treeprocessor(Treeprocessor):
+
     def run(self, root):
+
+        el_class_dict = {
+            "img": "img-fluid",
+            "blockquote": "blockquote",
+            "table": "table table-hover table-borderless",
+            "thead": "thead-light",
+            "code": "codehilite",
+        }
+
         for el in root.iter():
-            if el.tag == "img":
-                el.set("class", "img-fluid")
 
-            elif el.tag == "blockquote":
-                el.set("class", "blockquote")
+            if el.tag in el_class_dict:
+                BS4Treeprocessor.set_css_class(el, el_class_dict[el.tag])
+
+        for i in range(len(self.md.htmlStash.rawHtmlBlocks)):
+
+            bs4block = BeautifulSoup(
+                self.md.htmlStash.rawHtmlBlocks[i], "html.parser"
+            )
+
+            el = bs4block.find()
+
+            if el and el.name in el_class_dict:
+                BS4Treeprocessor.set_css_class(el, el_class_dict[el.name])
+                self.md.htmlStash.rawHtmlBlocks[i] = str(bs4block)
+
+    @staticmethod
+    def set_css_class(element, class_name: str):
+
+        if isinstance(element, ElementTree.Element):
+
+            current_class = element.attrib.get("class", "")
+
+            if class_name not in current_class:
+                new_class = f"{current_class} {class_name}".strip()
+                element.set("class", new_class)
+
+        elif isinstance(element, BeautifulSoup().new_tag("").__class__):
 
-            elif el.tag == "table":
-                el.set("class", "table table-hover table-borderless")
+            if "class" not in element.attrs:
+                element.attrs["class"] = []
 
-            elif el.tag == "thead":
-                el.set("class", "thead-light")
+            current_class = element["class"]
 
-            elif el.tag == "code":
-                el.set("class", "codehilite")
+            for name in class_name.split(" "):
+                if class_name not in current_class:
+                    current_class.append(name)
+        else:
+            raise TypeError(
+                "element can either be of type {}.{} or {}.{}".format(
+                    BeautifulSoup().new_tag("").__class__.__module__,
+                    BeautifulSoup().new_tag("").__class__.__qualname__,
+                    ElementTree.Element.__module__,
+                    ElementTree.Element.__qualname__,
+                )
+            )
```

Comment:
```suggestion
            raise TypeError("Unsupported element")
```

Code style - don't use `format`, use `f` strings instead. All the detail is unnecessary.

Reply:
Done